firefox-ios/Providers/RustSyncManager.swift

```diff
@@ -615,8 +615,10 @@ public class RustSyncManager: NSObject, SyncManager {
 
     @discardableResult
     public func syncEverything(why: SyncReason) -> Success {
-        return syncRustEngines(why: why,
-                               engines: syncManagerAPI.rustTogglableEngines.compactMap { $0.rawValue }) >>> succeed
+        return syncRustEngines(
```

Comment:
I think we should add a comment
```suggestion
        // Convert Deferred<Maybe<SyncResult>> into Deferred<Maybe<Void>>:
        // - If sync succeeds, return success with ().
        // - If sync fails, propagate the same failure.
        return syncRustEngines(
```